### PR TITLE
#156958165: Sort and Filtering search result

### PR DESCRIPTION
**Pull request by omokolataiwo**
#### What does this PR do?
Filters and sorts product search result.

#### Description of Task to be completed?
- Filter search result based on vendors and price
- Sort search and filtered result based on newest product, vendor, and price

#### How should this be manually tested?
- Clone the repository
- In the terminal run ```cd nazgul-rc```
- Run ```reaction``` in the terminal
- Open the browser and navigate to http://localhost:3000/
- Search for a product
- You can filter and sort from the search modal

#### Screenshots
screen shot 2018-05-28 at 10 24 42 am

screen shot 2018-05-27 at 4 23 37 pm

#### What are the relevant pivotal tracker stories?
[Finishes: #156958165]

### CODE REVIEW

File: e2eTest/sortFilter.test.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      .pause(500)
+      .saveScreenshot("./reports/snap/vendor-filter-result.png");
+  },
+  "User should be able to sort search result by newest product": browser => {
+    browser
+      .click("#vendor option:nth-child(2)")
+      .pause(500)
+      .saveScreenshot("./reports/snap/vendor-filter-result.png");
+  }
+  // "User should be able to sort search result by product price": browser => {
```
Comment (morekoya) [inline e2eTest/sortFilter.test.js]: Why are these commented out?

**Comment by morekoya** [comment]
Are you using the same font styling as the rest of the app? It looks a bit different.

File: e2eTest/sortFilter.test.js (hunk 1 above)
Comment (omokolataiwo) [reply e2eTest/sortFilter.test.js]: I created this pull request to create a review app and get feedback from the product owner. The test was not completed at this point.

File: imports/plugins/included/ui-search/client/templates/searchModal/searchFilter.html [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+          <label class="transform">Price</label>
+
+          <div class="rui select">
+
+            <select id="price-filter">
+
+              <option value="null" selected disabled>Filter by price</option>
+
+              <option value="all">All prices</option>
+
```
Comment (morekoya) [inline imports/plugins/included/ui-search/client/templates/searchModal/searchFilter.html]: Why do you have blank lines in-between?

File: imports/plugins/included/ui-search/lib/components/searchModal.js [hunk 3]
```diff
@@ -90,34 +115,103 @@ class SearchModal extends Component {
...
-              unmountMe={this.props.unmountMe}
-              isSearch={true}
-            />
-          }
-          {this.props.accounts.length > 0 &&
+          <div className="nazgul-sort-filter" style={{ fontSize: "2em", textAlign: "center" }}>
+            {this.renderFilter()}
+            {this.renderSort()}
+          </div>
+          {this.props.products.length > 0 && <ProductGridContainer products={this.props.products} unmountMe={this.props.unmountMe} isSearch={true} />}
```
Comment (morekoya) [inline imports/plugins/included/ui-search/lib/components/searchModal.js]: Isn't this line longer that it should be? How come it wasn't flagged, has hound been turned off?

**Review by morekoya — CHANGES_REQUESTED** [review]

**Review by morekoya — APPROVED** [review]

**Pull request merged**